### ISSUE TITLE
automatic exposure, the automatic gain

### ISSUE THREAD

**Issue by neo-dotcom** [RobboRob/allsky#2]
Time lapse with the high quality raspberry pi camera. First of all I would like to thank those who have rewritten Thomas' software! But what doesn't work is the automatic exposure, the automatic gain, which you have to set by hand. If you don't, the image will remain black.

**Comment by DarKOrange75** [comment]
Unfortunatly I think it is an issue with the raspberry software not Thomas/Rob software. If you take a picture with raspistill, the command line used by AllSky software, AE and AG will not go as far as they should and produce a black image.

**Comment by robertpascale** [comment]
I found the reason why Auto Exposure does not work with the RasPi cams on long exposure.

https://github.com/raspberrypi/userland/issues/632#issuecomment-636521905

To get long exposures working, effectively preview is disabled and burst mode activated, in order to stop having to wait for excessive amounts of time for the image to be taken. During the preview is when the stats for exposure and AWB etc are generated.

So, in a nutshell, unless some major things happen under the hood, we are going to be limited.
Only alternative would be to try to get the raw data and do everything yourself, but this seems like major overkill given the improvements that are likely to result wouldn't quite justify the effort.

**Comment by DarKOrange75** [comment]
Thanks Robert! You rule! Bad news for us but after all I am satisfied from what I get from this camera (thanks to you) for the price I paid.

**Comment by RobboRob** [comment]
I have taken the auto exposure switch out of the software to avoid the confusion...
